File: patterns/creational/builder/typescript/src/director.ts
import type { Builder } from "./builder.ts";

export class Director {
  buildMinimalViableProduct(builder: Builder): void {
    builder.buildPartA();
  }

  buildFullFeaturedProduct(builder: Builder): void {
    builder.buildPartA();
    builder.buildPartB();
    builder.buildPartC();
  }
}

File: patterns/creational/builder/csharp/src/Builder/Director.cs
namespace Builder;

public sealed class Director
{
    public void BuildMinimalViableProduct(IBuilder builder)
    {
        ArgumentNullException.ThrowIfNull(builder);
        builder.BuildPartA();
    }

    public void BuildFullFeaturedProduct(IBuilder builder)
    {
        ArgumentNullException.ThrowIfNull(builder);
        builder.BuildPartA();
        builder.BuildPartB();
        builder.BuildPartC();
    }
}
